src/main/scala/com/vngrs/etl/Loader.scala

```diff
@@ -0,0 +1,31 @@
+package com.vngrs.etl
+
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.Row
+
+trait Loader[A] {
+
+  protected[etl] def apply(input: RDD[A]): Loader[A]
+
+}
+
+object Loader {
+  def apply[A](f: RDD[A] => Unit): Loader[A] = new Loader[A]() {
+    override def apply(data: RDD[A]): Loader[A] = {
+      f(data)
+      this
+    }
+  }
+
+  def file(path: String): Loader[String] = new Loader[String] {
```

Comment:
That was just for demonstration actually. Since we are going to build this project as collection of minimal plugins, final design will be like what you wrote.